### ISSUE TITLE
The contents of a string gets formatted unexpectedly

### ISSUE THREAD

**Issue by davegallant** [nvuillam/npm-groovy-lint#64]
Running `--format` causes this change to occur:

```
diff --git a/Jenkinsfile b/Jenkinsfile
index c30cd86..c555b1a 100644
--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -13,7 +13,7 @@ def steps = {

   image.inside {
     sshagent(credentials: ['ssh-creds']) {
-      env.GIT_SSH_COMMAND='ssh -T -o StrictHostKeyChecking=no'
+      env.GIT_SSH_COMMAND = 'ssh -T -o StrictHostKeyChecking = no'

       stage('Validate Terraform') {
         sh 'terraform validate'
(END)
```

I would not expect this to be formatted as it causes the build to fail.

**Comment by nvuillam** [comment]
It seems to be a issue with the rule of SpaceAroundOperator, kind of false positive
Maybe it is solved in CodeNarc 1.6, but migration is not achieved yet

Meanwhile, several solutions :
- disable SpaceAroundOperator in the .groovylintrc.json associated to your project
- add some check in the [fix rule](https://github.com/nvuillam/npm-groovy-lint/blob/master/lib/rules/SpaceAroundOperator.js#L23) to use [notBetweenQuoteOrComments function] (https://github.com/nvuillam/npm-groovy-lint/blob/master/lib/utils.js#L338) and do not make the replacement if it fails

I see your forked the repo, so which solution do you prefer ?
- .groovylintrc.json then wait that i migrate to CodeNarc 1.6 ?
- solve the issue in your fork ?
- wait for me to solve the issue ?

**Comment by davegallant** [comment]
> It seems to be a issue with the rule of SpaceAroundOperator, kind of false positive
> Maybe it is solved in CodeNarc 1.6, but migration is not achieved yet
> 
> Meanwhile, several solutions :
> 
> * disable SpaceAroundOperator in the .groovylintrc.json associated to your project
> * add some check in the [fix rule](https://github.com/nvuillam/npm-groovy-lint/blob/master/lib/rules/SpaceAroundOperator.js#L23) to use [notBetweenQuoteOrComments function] (https://github.com/nvuillam/npm-groovy-lint/blob/master/lib/utils.js#L338) and do not make the replacement if it fails
> 
> I see your forked the repo, so which solution do you prefer ?
> 
> * .groovylintrc.json then wait that i migrate to CodeNarc 1.6 ?
> * solve the issue in your fork ?
> * wait for me to solve the issue ?

Thank you for the suggestions. I am unable to reproduce this now oddly. I will re-open if I can.

**Comment by nvuillam** [comment]
Maybe this upgrade impacted CodeNarc analysis, I added a test rule with your code and I can not reproduce either :D 

That's called luck 🤣

**Comment by davegallant** [comment]
I can reproduce this now. If codenarc detects a valid `SpaceAroundOperator` on the same line, it fixes the problem, but also goes into the string and fixes it there also causing breaking changes:

`env.GIT_SSH_COMMAND="ssh -T -o StrictHostKeyChecking=no"` -> `env.GIT_SSH_COMMAND = 'ssh -T -o StrictHostKeyChecking = no`

**Comment by nvuillam** [comment]
Solved again in latest patch 😁
Plz reopen if necessary :)
